library/core/src/num/mod.rs

```diff
@@ -782,13 +782,10 @@ macro_rules! from_str_radix_int_impl {
 from_str_radix_int_impl! { isize i8 i16 i32 i64 i128 usize u8 u16 u32 u64 u128 }
 
 macro_rules! doit {
-    ($($t:ty)*) => ($(impl FromStrRadixHelper for $t {
-        #[inline]
-        fn min_value() -> Self { Self::MIN }
-        #[inline]
-        fn max_value() -> Self { Self::MAX }
-        #[inline]
-        fn from_u32(u: u32) -> Self { u as Self }
+    ($sized:literal | $($t:ty)*) => ($(impl FromStrRadixHelper for $t {
+        const SIGNED: bool = $sized;
+        const ZERO: Self = 0 as Self;
```

Comment:
This cast shouldn't be necessary, the compiler will infer the type.
```suggestion
        const ZERO: Self = 0;
```